titiler/core/titiler/core/factory.py

```diff
@@ -410,14 +410,20 @@ def tilejson(
                 route_params["format"] = tile_format.value
             tiles_url = self.url_for(request, "tile", **route_params)
 
-            q = dict(request.query_params)
-            q.pop("TileMatrixSetId", None)
-            q.pop("tile_format", None)
-            q.pop("tile_scale", None)
-            q.pop("minzoom", None)
-            q.pop("maxzoom", None)
-            qs = urlencode(list(q.items()))
-            tiles_url += f"?{qs}"
+            qs_key_to_remove = [
+                "TileMatrixSetIt",
+                "tile_format",
+                "tile_scale",
+                "minzoom",
+                "maxzoom",
+            ]
+            qs = [
+                (key, value)
+                for (key, value) in request.query_params._list
```

Comment:
`dict(request.query_params)` will not keep the multiple inputs for `rescale` so we need to use `._list`.